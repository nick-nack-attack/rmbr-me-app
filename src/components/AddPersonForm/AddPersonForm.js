import React, { Component } from 'react';
import { Input, Button } from '../Utils/Utils';
import RmbrApiService from "../../services/rmbr-api-service";
import RmbrmeContext from "../../contexts/RmbrmeContext";

export default class AddPersonForm extends Component {

    constructor(props) {
        super(props);
        this.state = {
            formPersonName: '',
            selectedOption: 'Friend',
        };
    };

    static contextType = RmbrmeContext;

    handleInputChange = (event) => {
        const target = event.target;
        const value = target.value;
        const name = target.name;
        this.setState({
            [name]: value
        });
    };

    renderCategoryOptions() {
        const CategoryOptions = ['Friend', 'Co-Worker', 'Family'];
        return CategoryOptions.map((option, index) => {
            return (
                <label
                    htmlFor={option}
                    key={index}
                >
                    <input
                        name='selectedOption'
                        type='radio'
                        className='add_person_input'
                        value={option}
                        checked={this.state.selectedOption === option}
                        onChange={(e) => this.handleInputChange(e)}
                    />
                    {option}
                </label>
            );
        });
    };

    handleSubmit = (e) => {
        e.preventDefault();
        this.setState({ error: null });
        const person_name = this.state.formPersonName;
        const user_id = RmbrApiService.getUserId();
        const type_of_person  = this.state.selectedOption;
        const newPerson = { person_name, user_id, type_of_person };
        RmbrApiService.postPerson(newPerson)
            .then(res => {
                this.context.addPerson(res);
                this.setState({formPersonName: ''});
            })
            .catch(res => {
                this.setState({error: res.error});
            })
    };

    render() {
        return (
            <form
                className='add_person_form'
                onSubmit={(ev) => this.handleSubmit(ev)}
            >
                <legend>Add Person</legend>
                <div>
                    <Input
                        name='formPersonName'
                        id='personName'
                        placeholder='Type Person Here'
                        value={this.state.formPersonName}
                        onChange={(e) => this.handleInputChange(e)}
                    />
                </div>
                <div className='add_person_radio_options'>
                    {this.renderCategoryOptions()}
                </div>
                <div>
                    <Button
                        disabled={!this.state.formPersonName}
                    >
                        Submit
                    </Button>
                </div>
            </form>
        );
    };
};