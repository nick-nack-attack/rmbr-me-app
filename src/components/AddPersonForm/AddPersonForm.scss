@import '../../themes/v2';

.add_person_form {
    padding: 20px;
    max-width: 400px;
    margin-left: auto;
    margin-right: auto;
    border-radius: 10px;

    &>label {
        font-weight: bold;
    }

}

.multi-options {
    display: flex;
    justify-content: space-around;
}