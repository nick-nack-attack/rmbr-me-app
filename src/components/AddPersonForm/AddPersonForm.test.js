import React from "react";
import { shallow } from "enzyme";
import toJson from 'enzyme-to-json';
import AddPersonForm from "./AddPersonForm";

describe(`AddPersonForm component`, () => {
    it(`renders the complete form`, () => {
        const wrapper = shallow(<AddPersonForm/>)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
});