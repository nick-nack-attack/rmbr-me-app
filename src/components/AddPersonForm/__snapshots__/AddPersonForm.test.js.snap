// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddPersonForm component renders the complete form 1`] = `
<form
  className="add_person_form"
  onSubmit={[Function]}
>
  <legend>
    Add Person
  </legend>
  <div>
    <Input
      id="personName"
      name="formPersonName"
      onChange={[Function]}
      placeholder="Type Person Here"
      value=""
    />
  </div>
  <div
    className="add_person_radio_options"
  >
    <label
      htmlFor="Friend"
      key="0"
    >
      <input
        checked={true}
        className="add_person_input"
        name="selectedOption"
        onChange={[Function]}
        type="radio"
        value="Friend"
      />
      Friend
    </label>
    <label
      htmlFor="Co-Worker"
      key="1"
    >
      <input
        checked={false}
        className="add_person_input"
        name="selectedOption"
        onChange={[Function]}
        type="radio"
        value="Co-Worker"
      />
      Co-Worker
    </label>
    <label
      htmlFor="Family"
      key="2"
    >
      <input
        checked={false}
        className="add_person_input"
        name="selectedOption"
        onChange={[Function]}
        type="radio"
        value="Family"
      />
      Family
    </label>
  </div>
  <div>
    <Button
      className=""
      disabled={true}
    >
      Submit
    </Button>
  </div>
</form>
`;
